pkg/apis/extensions/validation/validation.go

```diff
@@ -607,3 +607,46 @@ func ValidateScale(scale *extensions.Scale) validation.ErrorList {
 
 	return allErrs
 }
+
+// ValidateConfigMapName can be used to check whether the given ConfigMap name is valid.
+// Prefix indicates this name will be used as part of generation, in which case
+// trailing dashes are allowed.
+func ValidateConfigMapName(name string, prefix bool) (bool, string) {
+	return apivalidation.NameIsDNSSubdomain(name, prefix)
+}
+
+// ValidateConfigMap tests whether required fields in the ConfigMap are set.
+func ValidateConfigMap(cfg *extensions.ConfigMap) validation.ErrorList {
+	allErrs := validation.ErrorList{}
+	allErrs = append(allErrs, apivalidation.ValidateObjectMeta(&cfg.ObjectMeta, true, ValidateConfigMapName, validation.NewFieldPath("metadata"))...)
+
+	if len(cfg.Data) == 0 {
```

Comment:
I had the same question in my head as well, but didn't get around to voicing it. I would say it's an unnecessary restriction and can be removed.